spec/museum_spec.rb

```diff
@@ -0,0 +1,196 @@
+
+            patron_3 = Patron.new("Johnny", 5)
+            patron_3.add_interest("Dead Sea Scrolls")
+
+            dmns.admit(patron_1)
+            dmns.admit(patron_2)
+            dmns.admit(patron_3)
+
+            expect(dmns.patrons_by_exhibit_interest).to eq({
+                gems_and_minerals => [patron_1],
+                dead_sea_scrolls => [patron_1, patron_2, patron_3],
+                imax => []
+            })
+        end
+    end
+
+    describe '#ticket_lottery_contestants' do
+        it 'returns patrons who are interested in an exhibit but do not have the spending money' do
+            dmns = Museum.new("Denver Museum of Nature and Science")
+            gems_and_minerals = Exhibit.new({name: "Gems and Minerals", cost: 0})
+            dead_sea_scrolls = Exhibit.new({name: "Dead Sea Scrolls", cost: 10})
+            imax = Exhibit.new({name: "IMAX",cost: 15})
+
+            dmns.add_exhibit(gems_and_minerals)
+            dmns.add_exhibit(dead_sea_scrolls)
+            dmns.add_exhibit(imax)
+
+            patron_1 = Patron.new("Bob", 0)
+            patron_1.add_interest("Gems and Minerals")
+            patron_1.add_interest("Dead Sea Scrolls")
+
+            patron_2 = Patron.new("Sally", 20)
+            patron_2.add_interest("Dead Sea Scrolls")
+
+            patron_3 = Patron.new("Johnny", 5)
+            patron_3.add_interest("Dead Sea Scrolls")
+
+            dmns.admit(patron_1)
+            dmns.admit(patron_2)
+            dmns.admit(patron_3)
+
+            expect(dmns.ticket_lottery_contestants(dead_sea_scrolls)).to eq([patron_1, patron_3])
+            expect(dmns.ticket_lottery_contestants(gems_and_minerals)).to eq([])
+            expect(dmns.ticket_lottery_contestants(imax)).to eq([])
+        end
+    end
+
+    describe '#draw_lottery_winner' do
+        it 'returns a random winner from the eligible lottery contestants' do
+            dmns = Museum.new("Denver Museum of Nature and Science")
+            gems_and_minerals = Exhibit.new({name: "Gems and Minerals", cost: 0})
+            dead_sea_scrolls = Exhibit.new({name: "Dead Sea Scrolls", cost: 10})
+            imax = Exhibit.new({name: "IMAX",cost: 15})
+
+            dmns.add_exhibit(gems_and_minerals)
+            dmns.add_exhibit(dead_sea_scrolls)
+            dmns.add_exhibit(imax)
+
+            patron_1 = Patron.new("Bob", 0)
+            patron_1.add_interest("Gems and Minerals")
+            patron_1.add_interest("Dead Sea Scrolls")
+
+            patron_2 = Patron.new("Sally", 20)
+            patron_2.add_interest("Dead Sea Scrolls")
+
+            patron_3 = Patron.new("Johnny", 5)
+            patron_3.add_interest("Dead Sea Scrolls")
+
+            dmns.admit(patron_1)
+            dmns.admit(patron_2)
+            dmns.admit(patron_3)
+
+            winner = dmns.draw_lottery_winner(dead_sea_scrolls)
+
+            expect([patron_1.name, patron_3.name]).to include(winner) #could have maybe stubbed here
```

Comment:
I actually don't know that I would have used a stub here - I think I would end up with what you have.  My reasoning would be that this methods 'purpose' is to randomly pick from the contestants.  I _could_ stub the contestants to be an array with only one option... but I think that doesn't allow me to really see the method working as it should.  Its a fine line, but I think what you have here is great!  

And, the stub that you created below is _exactly_ what I would have done.